src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/UnusedImportsCheck.java

```diff
@@ -231,14 +231,26 @@ private boolean isUnusedImport(String imprt) {
     private void processIdent(DetailAST ast) {
         final DetailAST parent = ast.getParent();
         final int parentType = parent.getType();
-        if (parentType != TokenTypes.DOT
+        if ((parentType != TokenTypes.DOT
             && parentType != TokenTypes.METHOD_DEF
             || parentType == TokenTypes.DOT
-                && ast.getNextSibling() != null) {
+                && ast.getNextSibling() != null)
+                && !isInnerInterfaceOrClass(parentType)) {
             referenced.add(ast.getText());
         }
     }
 
+    /**
+     * Checks whether the given parentType of the AST is an inner interface or inner class.
+     *
+     * @param parentType The token type
+     * @return true if the token type is INTERFACE_DEF or CLASS_DEF
+     */
+    private static boolean isInnerInterfaceOrClass(int parentType) {
+        return parentType == TokenTypes.INTERFACE_DEF
```

Comment:
Why this fix is for classes and interfaces only? There should be three violations:
```
package test;

import java.util.Set; // violation
import java.util.Map; // violation
import java.util.List; // violation

public class TestClass {

    @interface Map {}

    @Map
    enum Set {
        List(0);
        Set(int arg){}
    }

    TestClass() {
        Set x = Set.List;
    }
}
```

Reply:
@pbludov done.